File: clients/php/test/Auth/AccessTokenAuthContextTest.php
<?php declare(strict_types=1);

namespace Eternaltwin\Test\Auth;

use Eternaltwin\Auth\AccessTokenAuthContext;
use Eternaltwin\Auth\AuthScope;
use Eternaltwin\Oauth\OauthClientDisplayName;
use Eternaltwin\Oauth\OauthClientId;
use Eternaltwin\Oauth\OauthClientKey;
use Eternaltwin\Oauth\ShortOauthClient;
use Eternaltwin\Test\SerializationTestItem;
use Eternaltwin\User\ShortUser;
use Eternaltwin\User\UserDisplayName;
use Eternaltwin\User\UserDisplayNameVersion;
use Eternaltwin\User\UserDisplayNameVersions;
use Eternaltwin\User\UserId;
use PHPUnit\Framework\TestCase;

final class AccessTokenAuthContextTest extends TestCase {
  /**
   * @dataProvider provideFromJson
   * @param SerializationTestItem $item
   */
  public function testFromJson(SerializationTestItem $item): void {
    $actual = AccessTokenAuthContext::fromJson($item->getJson());
    $this->assertEquals($item->getValue(), $actual);
  }

  public function provideFromJson(): array {
    return SerializationTestItem::fromTestDir(
      "core/auth/access-token-auth-context",
      [
        "eternalfest-demurgos" => new AccessTokenAuthContext(
          AuthScope::Default(),
          new ShortOauthClient(
            OauthClientId::fromString("d19e61a3-83d3-410f-84ec-49aaab841559"),
            new OauthClientKey("eternalfest@clients"),
            new OauthClientDisplayName("Eternalfest"),
          ),
          new ShortUser(
            UserId::fromString("9f310484-963b-446b-af69-797feec6813f"),
            new UserDisplayNameVersions(new UserDisplayNameVersion(new UserDisplayName("Demurgos")), []),
          ),
        ),
      ],
    );
  }
}

File: clients/kotlin/src/test/kotlin/net/eternaltwin/auth/AccessTokenAuthContextTest.kt
package net.eternaltwin.auth;

import net.eternaltwin.SerializationTestItem
import net.eternaltwin.oauth.OauthClientDisplayName
import net.eternaltwin.oauth.OauthClientId
import net.eternaltwin.oauth.OauthClientKey
import net.eternaltwin.oauth.ShortOauthClient
import net.eternaltwin.user.*
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream
import kotlin.test.assertEquals

class AccessTokenAuthContextTest {
  companion object {
    @JvmStatic
    fun fromJsonString(): Stream<SerializationTestItem<AccessTokenAuthContext>> =
      SerializationTestItem.streamFromTestDir(
        "auth/access-token-auth-context",
        mapOf(
          "eternalfest-demurgos" to AccessTokenAuthContext(
            scope = AuthScope.Default,
            client = ShortOauthClient(
              id = OauthClientId("d19e61a3-83d3-410f-84ec-49aaab841559"),
              key = OauthClientKey("eternalfest@clients"),
              displayName = OauthClientDisplayName("Eternalfest"),
            ),
            user = ShortUser(
              id = UserId("9f310484-963b-446b-af69-797feec6813f"),
              displayName = UserDisplayNameVersions(UserDisplayNameVersion(UserDisplayName("Demurgos"))),
            )
          ),
        ),
      )
  }

  @ParameterizedTest
  @MethodSource
  fun fromJsonString(item: SerializationTestItem<AccessTokenAuthContext>) {
    val actual = AccessTokenAuthContext.fromJsonString(item.jsonString)
    assertEquals(item.value, actual)
  }
}
